File: leetcode/medium/1513. Number of Substrings With Only 1s.ts
function numSub(s: string): number {
  // or [1-nth] * ([1-nth] + 1) / 2
  let count: number = 0;
  let temp_count_1: number = 0;
  for (let i = 0; i <= s.length; i += 1) {
    if (i === s.length || s[i] !== "1") {
      count += (temp_count_1 * (temp_count_1 + 1)) / 2;
      temp_count_1 = 0;
      continue;
    }
    temp_count_1 += 1;
  }
  return count % (10 ** 9 + 7);

  // 1
  // 1 :: 1

  // 11
  // 12 :: 3

  // 111
  // 123 :: 6

  // below is the solution
  // let count: number = 0;
  // let temp_count: number = 1;
  // for (let i = 0; i <= s.length; i += 1){
  //     if (i === s.length || s[i] !== "1" ) {
  //         temp_count = 1;
  //         continue;
  //     }
  //     count += temp_count;
  //     temp_count += 1;
  // }
  // return count % (10**9+7);
}

File: leetcode/medium/1513. Number of Substrings With Only 1s.rs
impl Solution {
    pub fn num_sub(s: String) -> i32 {
        const MODULO: i64 = 10_i64.pow(9) + 7;
        let mut count: i64 = 0;
        let mut temp_count_1: i64 = 0;
        for (i, c) in s.chars().enumerate(){
            if c != '1'{
                count += (temp_count_1 * (temp_count_1+1)/2);
                temp_count_1 = 0;
                continue;
            }
            temp_count_1 += 1;
        }
        count += (temp_count_1 * (temp_count_1+1)/2);
        return (count % MODULO) as i32;
    }
}